notes/stv/20191120-openstack-deployment.txt

```diff
@@ -0,0 +1,761 @@
+cp /home/fedora/zeppelin/conf/shiro.ini.template /home/fedora/zeppelin/conf/shiro.ini
+
+
+## Set Zeppelin Configuration
+cp /home/fedora/zeppelin/conf/zeppelin-site.xml.template /home/fedora/zeppelin/conf/zeppelin-site.xml
+nano /home/fedora/zeppelin/conf/zeppelin-site.xml
+
+
+
+sudo su
+cat <<EOF >> "/etc/hosts"
+
+127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
+::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
+
+	
+10.0.0.15       stv-aglais-master.novalocal
+10.0.0.15       stv-aglais-master
+10.0.0.16       stv-aglais-worker-1
+10.0.0.9       stv-aglais-worker-2
+10.0.0.10       stv-aglais-worker-3
+10.0.0.5      stv-aglais-zeppelin
+
+EOF
+  
+exit
+
+
+cat > "${HOME:?}/.ssh/config" << EOF
+
+    Host stv-aglais-worker-1
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-worker-2 
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-worker-3
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-master.novalocal
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-master
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-zeppelin
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+EOF
+
+
+sudo chmod 600 ~/.ssh/config
+
+
+
+mkdir /home/fedora/spark
+
+## Copy Hadoop & Spark Config files from master
+ssh stv-aglais-master
+    scp ~/hadoop/etc/hadoop/* stv-aglais-master:/home/fedora/zeppelin/hadoop-conf;
+    scp -r ~/spark/* stv-aglais-zeppelin:/home/fedora/spark;
+exit
+
+
+
+## Enable Python3 and matplotlib for python3 notebooks
+## Update python interpreter, set it to python3
+
+## Install matplotlib
+sudo python3 -m pip install -U pip
+sudo python3 -m pip install -U matplotlib
+sudo python3 -m pip install -U healpy
+
+
+
+## Start Server
+/home/fedora/zeppelin/bin/zeppelin-daemon.sh start
+
+
+## Tunnel connection to 8080 and go to
+## On Local Machine (Browser)
+## Go to: http://localhost:8080/#/
+
+
+
+## Update Spark Interpreter from admin GUI
+## Set master=yarn-client
+
+
+
+
+#-------------------------------------------------------------------------
+#  Gaiasource & Tgas as Parquet files
+#-------------------------------------------------------------------------
+
+## From Gaia-spark virtual image (GDAF), copy ascii, parquet files from /archive/gaia/gdr1 to /hadoop/gaia/parquet & /hadoop/gaia/parquet on master node
```

Comment:
How do I do this - do I need to have a copy of the GDAF virtual machine running somewhere ?

Reply:
If you have downloaded the GDAF vm image file, then you could use [guestfs](http://libguestfs.org/) to extract the files without having to run the VM.